File: frontend/src/app/_model/assortment.ts
import {AssortmentUnit} from "./assortment-unit";
import {AssortmentCategory} from "./assortment-category";
import {AssortmentType} from "./assortment-type";

export interface Assortment {
  idAssortment?: number,
  name: string,
  quantity: number,
  volume: number,
  unit: AssortmentUnit,
  type: AssortmentType,
  category: AssortmentCategory,
}

File: backend/animedix/src/main/java/com/psleziona/animedix/model/Assortment.java
package com.psleziona.animedix.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.lang.NonNull;

import java.util.List;

@Entity
@Data
@NoArgsConstructor(force = true)
public class Assortment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer idAssortment;
    @NonNull
    private String name;
    private Double volume;
    private Integer quantity;
    private AssortmentUnit unit;
    private AssortmentType type;
    @NonNull
    private AssortmentCategory category;
    @OneToMany(mappedBy = "assortment")
    @JsonIgnore
    private List<UsedAssortment> usedInSurgeries;

}
